core/src/from_meta.rs

```diff
@@ -693,89 +694,106 @@ impl KeyFromPath for syn::Ident {
-                // An outer Err means no (key, value) structured could be found, while an Err in the
-                // second position of the tuple means that value was rejected by FromMeta.
-                //
-                // We defer key conversion into $key so that we don't lose span information in the case
-                // of String keys; we'll need it for good duplicate key errors later.
-                let pairs = nested
-                    .iter()
-                    .map(|item| -> Result<(&syn::Path, Result<V>)> {
-                        match *item {
-                            NestedMeta::Meta(ref inner) => {
-                                let path = inner.path();
-                                Ok((
-                                    path,
-                                    FromMeta::from_meta(inner).map_err(|e| e.at_path(&path)),
-                                ))
-                            }
-                            NestedMeta::Lit(_) => Err(Error::unsupported_format("expression")),
-                        }
-                    });
-
-                let mut errors = Error::accumulator();
-                // We need to track seen keys separately from the final map, since a seen key with an
-                // Err value won't go into the final map but should trigger a duplicate field error.
-                //
-                // This is a set of $key rather than Path to avoid the possibility that a key type
-                // parses two paths of different values to the same key value.
-                let mut seen_keys = HashSet::with_capacity(nested.len());
-
-                // The map to return in the Ok case. Its size will always be exactly nested.len(),
-                // since otherwise ≥1 field had a problem and the entire map is dropped immediately
-                // when the function returns `Err`.
-                let mut map = HashMap::with_capacity_and_hasher(nested.len(), Default::default());
-
-                for item in pairs {
-                    if let Some((path, value)) = errors.handle(item) {
-                        let key: $key = match KeyFromPath::from_path(path) {
-                            Ok(k) => k,
-                            Err(e) => {
-                                errors.push(e);
-
-                                // Surface value errors even under invalid keys
-                                errors.handle(value);
-
-                                continue;
-                            }
-                        };
+            map!(
+                HashMap::with_capacity_and_hasher($nested.len(), Default::default()),
+                $key,
+                $nested
+            );
+        }
+    };
 
-                        let already_seen = seen_keys.contains(&key);
+    (btree_map, $key:ty, $nested:ident) => {
+        impl<V: FromMeta + Ord> FromMeta for BTreeMap<$key, V> {
```

Comment:
Why does `V` have the `Ord` constraint here?

Reply:
I tested this, and found that the `Ord` constraint was unnecessary and created problems. I've removed it, and added a test for a non-`Ord` value.